spring-integration-core/src/main/java/org/springframework/integration/core/MessagingTemplate.java

```diff
@@ -97,4 +97,18 @@ public void setDefaultChannel(MessageChannel channel) {
 		return super.sendAndReceive(destination, requestMessage);
 	}
 
+	public Object receiveAndConvert(MessageChannel destination, long timeout) {
+		Message<?> message = doReceive(destination, timeout);
+		if (message != null) {
+			return doConvert(message, null);
```

Comment:
I am not sure what you mean; see `receive()` in `MessagingGatewaySupport` - he passes in `null` too.